OpenRA.Mods.Common/Scripting/Properties/TransportProperties.cs

```diff
@@ -34,13 +34,15 @@ public TransportProperties(ScriptContext context, Actor self)
 		public void LoadPassenger(Actor a) { cargo.Load(Self, a); }
 
 		[Desc("Remove the first actor from the transport.  This actor is not added to the world.")]
-		public Actor UnloadPassenger() { return cargo.Unload(Self); }
+		public Actor UnloadPassenger() { return cargo.UnloadLastEntered(Self); }
 
 		[ScriptActorPropertyActivity]
 		[Desc("Command transport to unload passengers.")]
 		public void UnloadPassengers()
 		{
-			Self.QueueActivity(new UnloadCargo(Self, true));
+			var toUnload = cargo.Passengers.ToList();
+			toUnload.Reverse();
```

Comment:
Then the `FirstInLastOutUnloading` flag on `Cargo` would be redundant. Policy should either be handled by the trait itself or by the callers, but not both. I very much prefer the former, with appropriate interfaces (ie. public methods and activities) to allow for flexibility.